source/dubregistry/internal/utils.d

```diff
@@ -13,3 +26,42 @@ string black(string url)
 @safe {
 	return black(URL(url)).toString();
 }
+
+/**
+ * Params:
+ *   file = the file to convert
+ *   deleteFinish = if true, delete the input file after a successful conversion
+ * Returns: the PNG stream of the icon or empty on failure
+ * Throws: Exception if name is empty or logo already exists and deleteExisting is not true
+ */
+bdata_t generateLogo(NativePath file) @safe
+{
+	static assert (isWeaklyIsolated!(typeof(&generateLogoUnsafe)));
+	static assert (isWeaklyIsolated!NativePath);
+	return (() @trusted => async(&generateLogoUnsafe, file).getResult())();
```

Comment:
This should to timeout after X seconds and terminate the `convert` process (at the moment it doesn't)